### PR TITLE
Added documentation for IfcRailwayPartTypeEnum

### PR DESCRIPTION
**Pull request by MatthieuPERIN**
Proposed documentation for Railway related new entities for Spatial description

* IfcRailwayPartTypeEnum

### CODE REVIEW

**Comment by MatthieuPERIN** [comment]
Need review by @TLiebich, @SergejMuhic and @czapplitec

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/Types/IfcRailwayPartTypeEnum/Documentation.md [hunk 1]
```diff
@@ -1 +1,21 @@
 The IfcRailwayPartTypeEnum defines the range of different types of railway part that can be specified.
+
+The Spatial structure of a Railway is sub-divided into several elements for lateral and vertical decomposition:
```
Comment (evandroAlfieri) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/Types/IfcRailwayPartTypeEnum/Documentation.md:3]: ...lateral, vertical and longitudinal

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/Types/IfcRailwayPartTypeEnum/Documentation.md [hunk 2]
```diff
@@ -1 +1,21 @@
...
+
+* TRACKSTRUCTURE is for the Track (vertical decomposition)
+* SUPERSTRUCTURE is for the upper part of the railway (vertical decomposition)
+* LINESIDESTRUCTURE is for the lateral part of the railway  (lateral decomposition)
```
Comment (evandroAlfieri) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/Types/IfcRailwayPartTypeEnum/Documentation.md:7]: couldn't LINESIDE be used for vertical breakdown too? I remember your picture
Picture1

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/Types/IfcRailwayPartTypeEnum/Documentation.md [hunk 3]
```diff
@@ -1 +1,21 @@
...
+
+!["Railway vertical and lateral spatial decomposition "](../../../../../../figures/IfcRailwayPartTypeEnum-global.png "Figure 1 &mdash; Railway decomposition")
+
+Railway track structure have longitudinal decomposition based on track specificities:
+
+* PLAINTRACKSUPESTRUCTURE for the plain line tracks (IfcElementAssemblyTypeEnum.TRACKPANEL).
+* DILATATIONSUPERSTRUCTURE for the area of Dilatation panels (IfcElementAssemblyTypeEnum.DILATATIONPANEL).
+* TURNOUTSUPERSTRUCTURE for the area of Turnouts (IfcElementAssemblyTypeEnum.TURNOUTPANEL).
+* TRACKSTRUCTUREPART for generic longitudinal decomposition if needed.
```
Comment (evandroAlfieri) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/Types/IfcRailwayPartTypeEnum/Documentation.md:16]: ...and for vertical. Again, according to the famous picture

**Review by czapplitec — CHANGES_REQUESTED** [review]
Suggestions for minor changes

**Review by SergejMuhic — COMMENTED** [review]
Not much to add to @evandroAlfieri and @czapplitec comments.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/Types/IfcRailwayPartTypeEnum/Documentation.md (hunk 2 above)
Comment (MatthieuPERIN) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/Types/IfcRailwayPartTypeEnum/Documentation.md:7]: @evandroAlfieri
Yes, but in the tabular shared with Infra it is presented as Lateral decomposition so I  limit the documentation to this aspect

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/Types/IfcRailwayPartTypeEnum/Documentation.md (hunk 3 above)
Comment (MatthieuPERIN) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/Types/IfcRailwayPartTypeEnum/Documentation.md:16]: @evandroAlfieri,

Again in the table shared with infra this lineside sub structure is flagged to be a longitudinal decomposition.

**Comment by MatthieuPERIN** [comment]
Asked changes have been applied, minor description improvement also done.

**Review by czapplitec — APPROVED** [review]

**Pull request merged**